geowebcache/nestedstorage/src/test/java/org/geowebcache/nested/NestedBlobStoreTest.java

```diff
@@ -0,0 +1,398 @@
+
+        store.get(to);
+        Mockito.verify(frontStore).get(to);
+        Mockito.verify(backingStore).get(to);
+        Mockito.verifyNoMoreInteractions(frontStore, backingStore);
+    }
+
+    @Test
+    public void testPut() throws StorageException {
+        TileObject to = configureQuery();
+        NestedBlobStore store = configureStore();
+
+        store.put(to);
+        Mockito.verify(frontStore).put(to);
+        Mockito.verify(backingStore).put(to);
+        Mockito.verifyNoMoreInteractions(frontStore, backingStore);
+    }
+
+    @Test
+    public void testGetLayerMetaDataFrontStore() throws StorageException {
+        TileObject to = configureQuery();
+        NestedBlobStore store = configureStore();
+
+        String key = "key";
+        String layerName = "layer";
+        Mockito.when(frontStore.getLayerMetadata(layerName, key)).thenReturn("value");
+
+        store.getLayerMetadata(layerName, key);
+        Mockito.verify(frontStore).getLayerMetadata(layerName, key);
+        Mockito.verifyNoMoreInteractions(frontStore, backingStore);
+    }
+
+    @Test
+    public void testGetLayerMetaDataBackingStoreIfFrontStoreEmpty() throws StorageException {
+        NestedBlobStore store = configureStore();
+
+        String key = "key";
+        String layerName = "layer";
+        Mockito.when(frontStore.getLayerMetadata(layerName, key)).thenReturn(null);
+        Mockito.when(backingStore.getLayerMetadata(layerName, key)).thenReturn("value");
+
+        store.getLayerMetadata(layerName, key);
+        Mockito.verify(frontStore).getLayerMetadata(layerName, key);
+        Mockito.verify(backingStore).getLayerMetadata(layerName, key);
+        Mockito.verify(frontStore).putLayerMetadata(layerName, key, "value");
+        Mockito.verifyNoMoreInteractions(frontStore, backingStore);
+    }
+
+    @Test
+    public void testPutLayerMetaData() throws StorageException {
+        NestedBlobStore store = configureStore();
+
+        String key = "key";
+        String layerName = "layer";
+        String value = "value";
+
+        store.putLayerMetadata(layerName, key, value);
+        Mockito.verify(frontStore).putLayerMetadata(layerName, key, value);
+        Mockito.verify(backingStore).putLayerMetadata(layerName, key, value);
+        Mockito.verifyNoMoreInteractions(frontStore, backingStore);
+    }
+
+    @Test
+    public void testDeleteTileObject() throws StorageException {
+        TileObject to = configureQuery();
+        NestedBlobStore store = configureStore();
+
+        store.delete(to);
+        Mockito.verify(frontStore).delete(to);
+        Mockito.verify(backingStore).delete(to);
+        Mockito.verifyNoMoreInteractions(frontStore, backingStore);
+    }
+
+    @Test
+    public void testDeleteTileRange() throws StorageException, MimeException {
+        long[][] rangeBounds = { //
+                { 0, 0, 0, 0, 0 }, //
+                { 0, 0, 1, 1, 1 }, //
+                { 0, 0, 3, 3, 2 } //
+        };
```

Comment:
I never have seen this before, using comments to prevent auto formatter from breaking wrapping. Nifty.